app-modules/portal/src/Http/Controllers/KnowledgeManagementPortal/KnowledgeManagementPortalArticleController.php

```diff
@@ -49,28 +49,32 @@ public function show(KnowledgeBaseCategory $category, KnowledgeBaseItem $article
     {
         $article->increment('portal_view_count');
 
-        return response()->json([
-            'category' => KnowledgeBaseCategoryData::from([
-                'id' => $category->getKey(),
-                'name' => $category->name,
-                'description' => $category->description,
-            ]),
-            'article' => KnowledgeBaseArticleData::from([
-                'id' => $article->getKey(),
-                'categoryId' => $article->category_id,
-                'name' => $article->title,
-                'lastUpdated' => $article->updated_at->format('M d Y, h:m a'),
-                'content' => tiptap_converter()->record($article, attribute: 'article_details')->asHTML($article->article_details),
-                'tags' => $article->tags()
-                    ->orderBy('name')
-                    ->select([
-                        'id',
-                        'name',
-                    ])
-                    ->get()
-                    ->toArray(),
-            ]),
-            'portal_view_count' => $article->portal_view_count,
-        ]);
+        if ($article->public === true) {
+            return response()->json([
+                'category' => KnowledgeBaseCategoryData::from([
+                    'id' => $category->getKey(),
+                    'name' => $category->name,
+                    'description' => $category->description,
+                ]),
+                'article' => KnowledgeBaseArticleData::from([
+                    'id' => $article->getKey(),
+                    'categoryId' => $article->category_id,
+                    'name' => $article->title,
+                    'lastUpdated' => $article->updated_at->format('M d Y, h:m a'),
+                    'content' => tiptap_converter()->record($article, attribute: 'article_details')->asHTML($article->article_details),
+                    'tags' => $article->tags()
+                        ->orderBy('name')
+                        ->select([
+                            'id',
+                            'name',
+                        ])
+                        ->get()
+                        ->toArray(),
+                ]),
+                'portal_view_count' => $article->portal_view_count,
+            ]);
+        }
+
+        return response()->json([]);
```

Comment:
You should refactor this to a negative check and an early return. Nesting the happy path can lead to complex and hard to read code:

```suggestion
        if (! $article->public) {
            return response()->json([], 401);
        }

        return response()->json([
                'category' => KnowledgeBaseCategoryData::from([
                    'id' => $category->getKey(),
                    'name' => $category->name,
                    'description' => $category->description,
                ]),
                'article' => KnowledgeBaseArticleData::from([
                    'id' => $article->getKey(),
                    'categoryId' => $article->category_id,
                    'name' => $article->title,
                    'lastUpdated' => $article->updated_at->format('M d Y, h:m a'),
                    'content' => tiptap_converter()->record($article, attribute: 'article_details')->asHTML($article->article_details),
                    'tags' => $article->tags()
                        ->orderBy('name')
                        ->select([
                            'id',
                            'name',
                        ])
                        ->get()
                        ->toArray(),
                ]),
                'portal_view_count' => $article->portal_view_count,
            ]);
```

Reply:
Done